LibSerialize.lua

```diff
@@ -958,7 +983,7 @@ LibSerialize._WriterTable = {
                 numAbs = -num
             end
             local asString = tostring(numAbs)
-            if #asString < 7 and tonumber(asString) == numAbs then
+            if #asString < 7 and tonumber(asString) == numAbs and IsFinite(numAbs) then
```

Comment:
Just as a pre-emptive clarification, the IsFinite test here is to handle an edge case where `math.huge` is serialized ingame, exported via saved variables, and then deserialized out of game. The reasoning being `tonumber("inf")` round trips back to `numAbs` ingame but not outside, so by explicitly disallowing non-finite values we force them to consistently serialize as NUM_FLOAT types at the expense of such values being a few bytes larger.

Reply:
Pre-emptive clarifications sound like a good place for comments :)

Also, do you see a clean way to avoid the tostring/tonumber round-trip for this case? Technically wasted effort if the number is `inf`.